lib/chef/provisioning/aws_driver/aws_tagger.rb

```diff
@@ -0,0 +1,56 @@
+require 'retryable'
+
+module Chef::Provisioning::AWSDriver
+# Include this module on a class or instance that is responsible for tagging
+# itself.  Fill in the hook methods so it knows how to tag itself.
+class AWSTagger
+  extend Forwardable
+
+  attr_reader :action_handler
+
+  def initialize(tagging_strategy, action_handler)
+    @tagging_strategy = tagging_strategy
+    @action_handler = action_handler
+  end
+
+  def_delegators :@tagging_strategy, :desired_tags, :current_tags, :set_tags, :delete_tags
+
+  def converge_tags
+    if desired_tags.nil?
+      Chef::Log.debug "aws_tags not provided, nothing to converge"
+      return
+    end
+
+    # Duplication and normalization
+    n_desired_tags = Hash[desired_tags.map {|k,v| [k.to_s, v.to_s]}]
+    n_current_tags = Hash[current_tags.map {|k,v| [k.to_s, v.to_s]}]
```

Comment:
What part is worth refactoring?  Changing all the keys/values into symbols?

For one-line changes like this I am okay without DRYing them up.  I think it can obfuscate the code to call a `keys_values_to_sym(desired_tags)` method here rather than just doing the operation inline.  Readers can quickly see what I am doing here where they have to go look up the definition of the `keys_values_to_sym` method if I refactor it.